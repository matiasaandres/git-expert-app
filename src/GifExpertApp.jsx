import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Goku"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) {
      return;
    }

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      {/* Titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}

      <AddCategory onNewCategory={(event) => onAddCategory(event)} />

      {/* Listado de Gif */}
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}

      {/*Item Gif */}
    </>
  );
};
