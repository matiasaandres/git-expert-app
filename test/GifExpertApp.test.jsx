import { render,screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"

describe('Prueba el component <GifExpertApp/>', () => { 

    test('Debe hacer mach con el snapshot', () => { 

       const {container} = render(<GifExpertApp/> );

       expect(container).toMatchSnapshot();

     });
     
 })