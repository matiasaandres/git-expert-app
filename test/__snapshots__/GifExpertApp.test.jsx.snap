// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba el component <GifExpertApp/> Debe  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      id=""
      name=""
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Prueba el component <GifExpertApp/> Debe hacer mach con el snap 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      id=""
      name=""
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Prueba el component <GifExpertApp/> Debe hacer mach con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      id=""
      name=""
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Prueba el component <GifExpertApp/> Debe hacer snapchop 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      id=""
      name=""
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`Prueba el component <GifExpertApp/> should first 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      id=""
      name=""
      placeholder="Buscar Gif"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
     Cargando... 
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
